### PR TITLE
Paul testing

### PR DESCRIPTION
**Pull request by paglenn**
updated testing file structure, frontend tests , and some backend ones. in case anyone else wants to work on testing or additional features, the structural changes are in summary:  

- features go in their own folder with associated components, helper functions, and hooks 
- component. tests in __tests__ inside component folder 
- backend tests in __tests__ in the same directory as whatever you are testing 

in short, each folder with modules has a __tests__ folder with tests for those modules. 
This way you always know where the tests are for whatever you're looking at and we don't have just one giant __tests__ folder with either a long list or a duplication of the file structure of the original .

### CODE REVIEW

**Review by davidmerid — APPROVED** [review]
We will need testing

**Review by BeseratT — APPROVED** [review]
Nice additions!
